android/sdl_android/src/main/java/com/smartdevicelink/managers/file/FileManager.java

```diff
@@ -81,100 +70,33 @@ public FileManager(ISdl internalInterface, Context context, FileManagerConfig fi
         this.context = new WeakReference<>(context);
     }
 
-    /**
-     * Creates and returns a PutFile request that would upload a given SdlFile
-     *
-     * @param file SdlFile with fileName and one of A) fileData, B) Uri, or C) resourceID set
-     * @return a valid PutFile request if SdlFile contained a fileName and sufficient data
-     */
     @Override
-    PutFile createPutFile(@NonNull final SdlFile file) {
-        PutFile putFile = new PutFile();
-        if (file.getName() == null) {
-            throw new IllegalArgumentException("You must specify an file name in the SdlFile");
```

Comment:
Filename should technically never be `null` because a name will be auto-generated in `SdlFile` if the dev set the name to `null`:
https://github.com/smartdevicelink/sdl_java_suite/blob/6c4a24d3b3e82a5b6c8299aa8b3b10f224cb64f2/android/sdl_android/src/main/java/com/smartdevicelink/managers/file/filetypes/SdlFile.java#L134

but I agree that it doesn't hurt to check that just in case.